src/neuroconv/datainterfaces/text/timeintervalsinterface.py

```diff
@@ -0,0 +1,63 @@
+from abc import abstractmethod
+
+import numpy as np
+from pynwb import NWBFile
+from pynwb.epoch import TimeIntervals
+
+from ...basedatainterface import BaseDataInterface
+from ...utils.types import FilePathType, Optional
+from ...tools.nwb_helpers import make_or_load_nwbfile
+
+
+def convert_df_to_time_intervals(df, name, description=None):
+    if description is None:
+        description = name
+    time_intervals = TimeIntervals(name, description)
+    if "start_time" not in df:
+        raise ValueError(f"df must contain a column named 'start_time'. Existing columns: {df.columns.to_list()}")
+    if "stop_time" not in df:
+        df["stop_time"] = np.nan
```

Comment:
Another design question - should we even allow a `TimeIntervals` mapping without some `stop_time` either available or mapping provided? It can sometimes be an indication that a `TimeIntervals` isn't the best representation of the data. Of course, still possible to be, but that leads to next suggestion...

...Or perhaps with an option to set the `stop_time` to be the the next consecutive `start_time` as we sometimes do when forming a trials table where there isn't really a definable stopping time except for when the next trial started?